src-cpp/rdkafkacpp_int.h

```diff
@@ -1103,7 +1103,8 @@ class ConsumerGroupMetadataImpl : public ConsumerGroupMetadata {
 class KafkaConsumerImpl : virtual public KafkaConsumer, virtual public HandleImpl {
 public:
   ~KafkaConsumerImpl () {
-
+    if (rk_)
+      rd_kafka_destroy(rk_);
```

Comment:
maybe you should set `RD_KAFKA_DESTROY_F_NO_CONSUMER_CLOSE`?

Reply:
Good call!